orangecontrib/text/corpus.py

```diff
@@ -228,6 +229,8 @@ def copy(self):
         """Return a copy of the table."""
         c = self.__class__(self.X, self.Y, self.metas, self.domain, self.text_features)
         c.ensure_copy()
+        c._tokens = copy.copy(self._tokens)
+        c._dictionary = copy.copy(self._dictionary)
```

Comment:
Since we are not using genism's `filter_extremes` or `compactify` methods I think the dictionary object itself does not change. So the answer is yes, we probably can consider it immutable which makes this copy redundant.

While I am playing around with this a bit I think I discovered an other problem. Preprocess widget should probably make a copy of the corpus. Currently it does not which causes some unintuitive scenarios like this one:
<img width="414" alt="screen shot 2016-07-01 at 23 19 05" src="https://cloud.githubusercontent.com/assets/713026/16534728/58581684-3fe2-11e6-8235-5d6c87322b2d.png">

IMO the left WordCloud should run default preprocessing since it was not passed through Preprocessing widget and the second should show tokens as they were preprocessed with Preprocessing text widget. Do you agree? Can you make a PR for this?